assets/app/scripts/services/apiEndpoint_v2.js

```diff
@@ -0,0 +1,311 @@
+'use strict';
+/* jshint unused:false */
+
+// see gist: https://gist.github.com/benjaminapetersen/540ac605d3c1b0660062
+// for discussion leading up to the creation of this file
+
+angular.module('openshiftConsole')
+  .factory('ApiEndpointService_v2', function(API_CFG) {
+
+    // defaultVersion lookup for backwards compatibility...
+    var defaultVersionByGroup = {
+      'extensions':   'v1beta1',
+      'experimental': 'v1beta1',
+      '':             'v1'  // this group applies to openshift & k8s (/oapi) & (/api)
+    };
+
+    // private processor
+    var mapApis = function(groupVersions) {
+      return _.reduce(
+              groupVersions,
+              function(memo, next) {
+                if(!memo[next.group]) {
+                  memo[next.group] = {};
+                }
+                memo[next.group].defaultVersion = defaultVersionByGroup[next.group];
+                if(!memo[next.group][next.version]) {
+                  memo[next.group][next.version] = {};
+                }
+                _.each(next.resources, function(resource) {
+                  if(!memo[next.group][next.version][resource]) {
+                    memo[next.group][next.version][resource] = {
+                      hostPort: next.hostPort,
+                      prefix: next.prefix,
+                      // for convenience, will include these fields in the object
+                      // even though they are used for lookup.
+                      group: next.group,
+                      version: next.version
+                    };
+                  }
+                });
+                return memo;
+              }, {});
+    };
+
+    // temporary, until we get a new OPENSHIFT_CONFIG w/the new groupVersion structure
```

Comment:
go ahead and change the config.js to the structure you expect, and work with it in `hack/serve-local-assets.sh`... I want a clean switch... not code handling both formats in master at the same time